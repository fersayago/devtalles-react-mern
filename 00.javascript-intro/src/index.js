// los ejercicios se encuentran en la carpeta 'bases'