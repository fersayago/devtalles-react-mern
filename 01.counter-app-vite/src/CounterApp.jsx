import { useState } from 'react'
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
  
  const [ counter, setCounter ] = useState(value);

  const handlePlusOne = (e) => setCounter( counter + 1 );

  const handleMinusOne = (e) => setCounter( counter -1 );

  const handleReset = (e) => setCounter( value );

  return(
    <>
      <h1>CounterApp</h1>
      <h2>{ counter }</h2>

      <button onClick={handleMinusOne}>
        -1
      </button>
      <button onClick={handlePlusOne}> {/* (event) => handlePlusOne(event) */}
        +1
      </button>
      <button onClick={handleReset}>
        Reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp;