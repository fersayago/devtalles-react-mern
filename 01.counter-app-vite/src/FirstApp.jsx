// si no se cambia el valor y no se tiene relaciÃ³n con ningun hook, conviene ponerlo por fuera del compoente funcional
// no se pueden pasar objetos para renderizar
const newMessage = {
  message: 'Bienvenido',
  name: 'Fernando'
}

const getGreeting = (nombre, mensaje) => {
  return `${mensaje} ${nombre}`
}

const FirstApp = () => {

  
  
  return (
    <>
    {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <h1>{ getGreeting(newMessage.name, newMessage.message) }</h1>
      <h2>Soy un subtitulo</h2>
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Illo, error esse labore magnam possimus est quo. Minima amet maxime, veniam ducimus molestiae et rerum, blanditiis, ipsam accusantium nesciunt id officia.</p>
    </>
  )
}

export default FirstApp