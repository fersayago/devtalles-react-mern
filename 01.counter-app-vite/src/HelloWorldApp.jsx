

export default function HelloWorldApp() {
  return (
    <h1>Hola mundo!!!</h1>
  );
};