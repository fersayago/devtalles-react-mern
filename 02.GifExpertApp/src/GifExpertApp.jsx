import { useState } from "react";

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Ciudad de Dios', 'Batman']);

  return (
    <>
      {/* Titulos */}
      <h1>GifExpertApp</h1>
      {/* Input */}
      {/* Listado de Gif */}
      <ol>
        {
          categories.map((category) => {
            return <li key={ category }>{category}</li>
          })
        }
      </ol>
        {/* Gif Item */}
    </>
  )
}

export default GifExpertApp