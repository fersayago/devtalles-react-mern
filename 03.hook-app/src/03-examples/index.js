export { default as MultipleCustomHooks } from './MultipleCustomHooks';
export { default as Quote } from './Quote';
export { default as QuoteLoader } from './QuoteLoader';
