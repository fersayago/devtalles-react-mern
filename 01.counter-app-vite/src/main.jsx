import React from 'react'
import ReactDOM from 'react-dom/client'
import FirstApp from './FirstApp';
import './styles.css';

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    <FirstApp titulo='Titulo por Props' /* texto='Texto desde props' *//>
  </React.StrictMode>
)