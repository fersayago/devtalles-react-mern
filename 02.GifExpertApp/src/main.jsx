import React from 'react'
import ReactDOM from 'react-dom/client'
import GifExpertApp from './GifExpertApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
)

// APIKEY=0yozM6lsSlv4u6HCwUqaLP4GJBbWPLF1