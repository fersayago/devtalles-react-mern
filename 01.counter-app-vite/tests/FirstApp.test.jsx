import { getByTestId, render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {

  test('debe de hacer match con el snapshot', () => {
    // nos referimos a que le tomamos una fotografia al componente y si
    // algo llega a cambiar, la comparaci√≥n falla

    // renderiza el componente en memoria
    const title = 'Titulo desde test'

    const { container } = render( <FirstApp title={ title } /> )
    expect( container ).toMatchSnapshot();
    
  })

  test('title en h1', () => {
    // se debe mostrar el titulo en h1 en el snapshot

    // primer revisamos que renderice el titulo
    const title= 'Titulo a evaluar'
    const { container, getByText, getByTestId } = render (<FirstApp title={ title }/>)
    expect( getByText(title) ).toBeTruthy()

    // // buscamos el elemento con querySelector y nos fijamos que contenga el titulo a buscar
    // const h1 = container.querySelector('h1')
    // expect(h1.innerHTML).toContain(title)

    // buscamos el titulo mediante el testId en caso de que no este en H1
    expect( getByTestId('test-title').innerHTML ).toContain(title);
  })

  test('Debe mostrar el subtitulo pasado por props', () => {
    const title = 'Titulo por test'
    const testSubtitle = 'Subtitulo por test'
    const { getAllByText } = render (
      <FirstApp title={ title } subTitle={ testSubtitle }/>
    )
    
    // nos fijamos que renderice el subtitulo
    expect( getAllByText(testSubtitle).length ).toBe(2);
  })
})