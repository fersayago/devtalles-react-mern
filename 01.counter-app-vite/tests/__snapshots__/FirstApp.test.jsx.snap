// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo desde test
  </h1>
  <p>
    Default Title
  </p>
  <p>
    Default Title
  </p>
  <p>
    Fernando Sayago
  </p>
</div>
`;
